driver.py

```diff
@@ -25,6 +25,59 @@
         sys.stdout.write(f'\rConnecting to DB ... {wait_time}')
 
 
+def reconnectDB():
+    """
+    Function to reconnect to the database if connection is closed for some reason
+    Args:   None
+    Return: None
+    """
+    wait_time = 0
+    global pgSqlConn, pgSqlCur
+    global is_connected
+    is_connected = False
+    while not is_connected:
+        try:
+            pgSqlCur,pgSqlConn = c.pg_connect()
+            is_connected = True
+            return is_connected
+        except:
+            time.sleep(1)
+            wait_time+=1
+            if wait_time == 30:
+                return False
+
+
+def check_conn():
+    """
+    Function to check the connection to the database and reconnect if closed
+    Args:  None
+    Return: None
+    """
+    if not is_connected:
+        return reconnectDB()
+    # check to make sure that the connection is open and active
+    try:
+        pgSqlCur.execute('SELECT 1')
+        return True
+    except:
+        return reconnectDB()
+
+
+def sql_except(err):
+    """
+    Function to print out the error message generated from the exception
+    Args:
+        err - the error message generated
+    Returns None
+    """
+    # get the details for exception
+    err_type, err_obj, traceback=sys.exc_info()
```

Comment:
Are we doing anything with this info?

Reply:
I was using it for making sure the exception was being caught on my ide console.  We can send it back to the server to display to user if you want.